2 feature step 2

korzhuk-xeniya opened the pull request.

Lebedeva-Elena committed:
добавила сущность объявлений.доведу до ума связи, как будут другие таблички наглядно. пока горит красным

korzhuk-xeniya committed:
создан CreateOrUpdateComment
исправлены CommentDto, Comments

korzhuk-xeniya committed:
CommentsAdsController связан с dto: CreateOrUpdateComment, CommentDto, Comments

korzhuk-xeniya committed:
Создана сущность Comment

korzhuk-xeniya committed:
Подключена база данных для Image and Comment

Kseniia313 committed:
Добавила API описание для контроллеров, добавила в pom зависимость validator для валидации данных

korzhuk-xeniya committed:
Добавлен CommentMapper

AlexandraGlinka committed:
User Entity

korzhuk-xeniya committed:
Добавлен пустой CommentServiceImpl

korzhuk-xeniya committed:
Добавлен CommentRepository

korzhuk-xeniya committed:
Initial commit

AlexandraGlinka committed:
Merge remote-tracking branch 'origin/1-feature-step_2-entity-aglinka' into 2-feature-step-2

Kseniia313 committed:
Добавила API описание для контроллеров, добавила в pom зависимость validator для валидации данных

Lebedeva-Elena committed:
закоммитила красноту

Kseniia313 committed:
Merge remote-tracking branch 'origin/Bystrova-Kseniia-1-dto-and-controllers' into 2-feature-step-2

Lebedeva-Elena committed:
Merge remote-tracking branch 'origin/eleb-AdEntity' into 2-feature-step-2

Lebedeva-Elena committed:
Merge remote-tracking branch 'origin/2-feature-step-2' into 2-feature-step-2

korzhuk-xeniya committed:
Убрала из пом плагин для mapstructure

korzhuk-xeniya committed:
Правки

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/1-step_2_xeny_k' into 2-feature-step-2

Lebedeva-Elena committed:
sql ads

AlexandraGlinka committed:
CommentDto изменен тип для поля createdAt

AlexandraGlinka committed:
User - ограничения дла полей

AlexandraGlinka committed:
NewPasswordDto - ограничения дла полей

AlexandraGlinka committed:
UserDto, UpdateUserDto - ограничения дла полей

AlexandraGlinka committed:
UserMapper - без маппера для картинки

AlexandraGlinka committed:
добавлено user.sql

AlexandraGlinka committed:
добавлено user.sql

AlexandraGlinka committed:
mapstruct

AlexandraGlinka committed:
CommentDto fix

AlexandraGlinka committed:
changelog добавила users

AlexandraGlinka committed:
UserMapper косметические правки

korzhuk-xeniya committed:
Добавлен ImageController

korzhuk-xeniya committed:
Добавлен UserOrAdminService
Дописан CommentServiceImpl
Поправлен CommentsAdsController

korzhuk-xeniya committed:
поменяла названия методов в CommentMapper

AlexandraGlinka committed:
Merge remote-tracking branch 'origin/2-feature-photomapper-xeniya_k' into 2-feature-user-entity-aglinka

korzhuk-xeniya committed:
добавлены логи в CommentServiceImpl

AlexandraGlinka committed:
Revert "UserMapper косметические правки"

This reverts commit a5e96ab3e8c4569d72aad708afeaae68a2b4522f.

ElenaZherebtsova committed:
Добавила CommentMapper1.

ElenaZherebtsova committed:
Добавила в маппер формат даты и в CommentDto поле даты сделала String.

korzhuk-xeniya committed:
Исправила на Integer поле createdAt

korzhuk-xeniya committed:
Исправила на Integer поле createdAt

AlexandraGlinka committed:
UserMapper доработка маппера для картинок

korzhuk-xeniya committed:
Закомментировала CommentMapper

AlexandraGlinka committed:
Merge remote-tracking branch 'origin/2-feature-user-entity-aglinka' into 2-feature-step-2

# Conflicts:
#	src/main/java/ru/skypro/homework/dto/CommentDto.java
#	src/main/java/ru/skypro/homework/mapper/CommentMapper.java

ElenaZherebtsova committed:
Merge branch '2-feature-ElenaZH' into 2-feature-step-2

# Conflicts:
#	src/main/java/ru/skypro/homework/dto/CommentDto.java
#	src/main/java/ru/skypro/homework/mapper/CommentMapper.java

ElenaZherebtsova committed:
Добавила в маппер формат даты и в CommentDto поле даты сделала String.

ElenaZherebtsova committed:
Удалила CommentMapper1.

korzhuk-xeniya committed:
Удален класс CommentMapper1

Lebedeva-Elena committed:
sql ads

Lebedeva-Elena committed:
Merge remote-tracking branch 'origin/eleb-AdsMapperAndSQL' into 2-feature-step-2

# Conflicts:
#	src/main/resources/liquibase/changelog-master-2.yml
#	src/main/resources/liquibase/scripts/users.sql

korzhuk-xeniya committed:
ИСПРАВЛЕНЫ ОШИБКИ ПОСЛЕ МЕРЖА

Lebedeva-Elena committed:
mapper ads

korzhuk-xeniya committed:
создан  AdsServiceImpl и AdsService

korzhuk-xeniya committed:
поправлена зависимость в AdsMapper
поправлен метод adsToAdsDto, остальное пока закомментировала, чтобы запустить проект

korzhuk-xeniya committed:
поправлена зависимость в AdsMapper
поправлен метод adsToAdsDto, остальное пока закомментировала, чтобы запустить проект

korzhuk-xeniya committed:
поправлена зависимость в AdsMapper
поправлен метод adsToAdsDto, остальное пока закомментировала, чтобы запустить проект

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/2-feature-ads-service-xeny_k' into 2-feature-step-2

korzhuk-xeniya committed:
внесены правки после мержа

OlgaLugacheva commented on src/main/resources/application.properties:
```diff
@@ -1,8 +1,8 @@
+spring.jpa.hibernate.ddl-auto=validate
+spring.liquibase.change-log=classpath:liquibase/changelog-master-2.yml
 
 server.port=8081
```
порт лучше 8080 иначе могут быть проблемы с фронтом

OlgaLugacheva commented on src/main/java/ru/skypro/homework/mapper/AdsMapper.java:
```diff
@@ -0,0 +1,36 @@
+
+@Mapper(componentModel = "spring")
+public interface AdsMapper {
+    AdsMapper INSTANSE = Mappers.getMapper(AdsMapper.class);
```
давайте красиво чтобы было - инжектить будем в сервисе(контроллере) а это удалите

OlgaLugacheva replied:
в остальных местах также поправить

OlgaLugacheva commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -1,69 +1,140 @@
-private final ImageService imageService;
+    private final ImageService imageService;
+    private final AdsService adsService;
+
```
private final AdsMapper adsMapper;

и в остальных случаях также

OlgaLugacheva commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -0,0 +1,51 @@
+@RequiredArgsConstructor
+@Slf4j
+public class AdsServiceImpl implements AdsService {
+    private AdsRepository adsRepository;
+    private UserRepository userRepository;
+    private ImageService imageService;
```
также не заинжектиться 
надо
private final

OlgaLugacheva replied:
провеоить в остальных сервисах и контроллерах

OlgaLugacheva commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -0,0 +1,51 @@
+        try {
+            image = imageService.saveImageFile(imageFile);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
ну точно не Runtime - придумайте кастомное исключение

OlgaLugacheva commented on src/main/java/ru/skypro/homework/entity/Ads.java:
```diff
@@ -0,0 +1,53 @@
+    private Integer price;
+    @Column(name = "TITLE")
+    private String title;
+
```
а description где?

OlgaLugacheva commented on src/main/java/ru/skypro/homework/entity/Ads.java:
```diff
@@ -0,0 +1,53 @@
+
+@Entity
+@Data
+//@Getter
+//@Setter
+//@ToString
+//@RequiredArgsConstructor
```
закомментированный явно лишний код убираем

OlgaLugacheva commented on src/main/java/ru/skypro/homework/entity/User.java:
```diff
@@ -0,0 +1,46 @@
+    private String password;
+
+    @Column(name = "REG_DATE", nullable = false)
+    private LocalDateTime regDate;
```
может быть для дат проще Timestamp ? Но решать вам

OlgaLugacheva commented on src/main/java/ru/skypro/homework/mapper/CommentMapper.java:
```diff
@@ -0,0 +1,31 @@
+@Mapper(componentModel = "spring")
+public interface CommentMapper
+{
+    CommentMapper INSTANSE = Mappers.getMapper(CommentMapper.class);
```
см выше

AlexandraGlinka committed:
USER - изменен тип для полей RegDate и Role

AlexandraGlinka committed:
обработан метод setPassword в сервисе UserService

korzhuk-xeniya committed:
Учтены замечания, AdsMapper доведен до состояния компиляции

korzhuk-xeniya committed:
Удален закомментированный код

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/2-feature-ads-xeny_k' into 2-feature-step-2

korzhuk-xeniya committed:
поправлены exceptions

korzhuk-xeniya committed:
добавлен final в ImageServiceImpl

el-kg committed:
Написаны методы getAllUserAds(),updateImageAd() для AdsService/AdsServiceImpl.
Эти же методы дописаны в AdsController'е.
В AdsRepository добавлен метод findByUser.
Сделал рефакторинг AdsNotFoundException,UserNotFoundException

AlexandraGlinka committed:
добавлен метод getAuthUserInfo в UserService, тест addCommentToAds_ закомментировала, т.к. изменен метод сервиса

AlexandraGlinka committed:
добавлен метод updateAuthUserInfo в UserService

AlexandraGlinka committed:
добавлены логеры для пользователя

AlexandraGlinka committed:
метод updateAvatar не доработан

AlexandraGlinka committed:
добавлен тестовый пользователь в users.sql

AlexandraGlinka committed:
изменен метод смены пароля (пока не работает)

AlexandraGlinka committed:
поправила user.sql (сделала дефолтную дату), добавила маппер для регистрации (возможно не нужен)

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/2-feature-exceptions-xeny_k' into 2-feature-step-2

korzhuk-xeniya committed:
сделаны правки комментарии отображаются

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/2-feature-ads-kgalkin-2metodsAdd' into 2-feature-step-2

# Conflicts:
#	src/main/java/ru/skypro/homework/exception/AdsNotFoundException.java
#	src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java

korzhuk-xeniya committed:
Merge remote-tracking branch 'origin/2-feature-правки' into 2-feature-step-2

AlexandraGlinka committed:
исправлен маппер и тест для пользователей

korzhuk-xeniya committed:
аватар обновляется

korzhuk-xeniya committed:
настроен mvcMatchers для получения объявлений без авторизации

korzhuk-xeniya committed:
удален явно не нужный закомментированный код

korzhuk-xeniya committed:
отображается иконка пользователя в комментарии
